src/pair.c

```diff
@@ -551,111 +551,156 @@ int get_pair_osc_arg_index(char* varname, char* oscargs, uint8_t argc, uint8_t s
+            else if(match_op(s1,'+'))
             {
                 offset += b;
             }
-            else if(strchr(s1,'-'))
+            else if(match_op(s1,'-'))
             {
                 offset += b;
                 scale *= -1;
             }
+            else
+            {
+                printf("\nERROR -failed to parse '%s'! nonsensical operator?\n", pre);
+                return -1;
+            }
             break;
         default:
-            if(strchr(pre,'-'))
+            if(match_op(pre,'-'))
             {
                 scale *= -1;
             }
-            else
+            else if (!is_ws(pre))
             {
-                //if its not whitespace, its nonsensical, we'll ignore it
+                printf("\nERROR -failed to parse '%s'! nonsensical operator?\n", pre);
+                return -1;
             }
             break;
         }//switch
     }//if pre conditions
-    if(strlen(post))
+    if(*post)
     {
         char s1[20],s2[20];
         float a,b;
-        switch(sscanf(post,"%[-+*/ ]%f%[+- ]%f",s1,&a,s2,&b))
+        switch(sscanf(post,"%[-+*/ ]%f%n%[+- ]%n%f%n",s1,&a,&end,s2,&end,&b,&end))
         {
         case 4:
-            if(strchr(s2,'+'))//only add/subtract makes sense here
+            if(match_op(s2,'+'))//only add/subtract makes sense here
             {
                 offset += b;
             }
-            else if(strchr(s2,'-'))
+            else if(match_op(s2,'-'))
             {
                 offset -= b;
             }
             else
             {
-                printf("\nERROR -could not get post conditioning! nonsensical operator?\n");
+                printf("\nERROR -failed to parse '%s'! nonsensical operator?\n", post);
                 return -1;
             }
         case 3:
-        //if its not whitespace, its nonsensical, we'll ignore it
+        //if its not whitespace, its nonsensical, will be caught below
         case 2:
-            if(strchr(s1,'*'))
+            if (!is_ws(post+end)) //don't allow trailing garbage
+            {
+                printf("\nERROR -failed to parse '%s'! nonsensical operator?\n", post);
+                return -1;
+            }
+            if(match_op(s1,'*'))
             {
                 scale *= a;
             }
-            else if(strchr(s1,'/'))
+            else if(match_op(s1,'/'))
             {
                 scale /= a;
             }
-            else if(strchr(s1,'+'))
+            else if(match_op(s1,'+'))
             {
                 offset += a;
             }
-            else if(strchr(s1,'-'))
+            else if(match_op(s1,'-'))
             {
                 offset -= a;
             }
+            else
+            {
+                printf("\nERROR -failed to parse '%s'! nonsensical operator?\n", post);
+                return -1;
+            }
             break;
         default:
-            //if its not whitespace, its nonsensical, we'll ignore it
```

Comment:
Do you mean that I should add a comment there in the source? Or comment here?

Anyway, in the default case of the switch, we couldn't parse the post conditioning. If it's just just whitespace then it's ok, we'll just ignore it. But if it isn't, then there's some trailing garbage (such as a trailing operator). The following if statement makes sure that the latter case is detected and an appropriate error message given.